jobrunner/controller/main.py

```diff
@@ -175,7 +175,7 @@ def handle_job(job, mode=None, paused=None):
     if job.cancelled:
         with transaction():
             cancel_job(job)
-            mark_job_as_failed(job, StatusCode.CANCELLED_BY_USER, "Cancelled by user")
+            set_code(job, StatusCode.CANCELLED_BY_USER, "Cancelled by user")
```

Comment:
very slight tingles that this change doesn't need a change in the tests (as was needed for `DEPENDENCY_FAILED`) - is there a tiny gap in testing coverage?